plugins/scaffolder-backend/src/service/helpers.ts

```diff
@@ -107,3 +107,35 @@ export async function findTemplate(options: {
 
   return template as TemplateEntityV1beta3;
 }
+
+/**
+ * Takes a single unknown parameter and makes sure that it's a single string or
+ * an array of strings, and returns as an array.
+ */
+export function parseStringsParam(
+  param: unknown,
+  paramName: string,
+): string[] | undefined {
+  if (param === undefined) {
+    return undefined;
+  }
+
+  const array = [param].flat();
+  if (array.some(p => typeof p !== 'string')) {
+    throw new InputError(`Invalid ${paramName}, not a string`);
+  }
+
+  return array as string[];
+}
+
+export function parseNumberParam(param: unknown, paramName: string) {
```

Comment:
Though these functions are trivial, would be nice to have some unit tests for them to cover happy path and corner cases